homeassistant/components/notion/binary_sensor.py

```diff
@@ -28,17 +33,45 @@
     SENSOR_WINDOW_HINGED_VERTICAL,
 )
 
-BINARY_SENSOR_TYPES = {
-    SENSOR_BATTERY: ("Low Battery", "battery"),
-    SENSOR_DOOR: ("Door", DEVICE_CLASS_DOOR),
-    SENSOR_GARAGE_DOOR: ("Garage Door", "garage_door"),
-    SENSOR_LEAK: ("Leak Detector", DEVICE_CLASS_MOISTURE),
-    SENSOR_MISSING: ("Missing", DEVICE_CLASS_CONNECTIVITY),
-    SENSOR_SAFE: ("Safe", DEVICE_CLASS_DOOR),
-    SENSOR_SLIDING: ("Sliding Door/Window", DEVICE_CLASS_DOOR),
-    SENSOR_SMOKE_CO: ("Smoke/Carbon Monoxide Detector", DEVICE_CLASS_SMOKE),
-    SENSOR_WINDOW_HINGED_HORIZONTAL: ("Hinged Window", DEVICE_CLASS_WINDOW),
-    SENSOR_WINDOW_HINGED_VERTICAL: ("Hinged Window", DEVICE_CLASS_WINDOW),
+BINARY_SENSOR_DESCRIPTIONS: dict[str, BinarySensorEntityDescription] = {
```

Comment:
One idea of the `EntityDescription` change was to refactor dictionary mappings if possible to just tuples.
Here, it would probably only require changing the for loop / list comprehension to also iterate over all descriptions and then checking if the key matches.

Reply:
Aren't `dict` lookups faster than iterating over an iterable?

Reply:
> Aren't `dict` lookups faster than iterating over an iterable?

They are. It won't make much of a difference here though as there are  only 10 items and the code is looped over just once. The benefit in using a `tuple` here is that you don't need to duplicate the key.

The list comprehension would only need to change slightly from
```py
            for task_id, task in coordinator.data["tasks"].items()
            if task["task_type"] in BINARY_SENSOR_DESCRIPTIONS
```

to
```py
            for task_id, task in coordinator.data["tasks"].items()
            for description in BINARY_SENSOR_DESCRIPTIONS
            if description.key in task["task_type"]
```

Reply:
Another benefit: With the modified list comprehension, you can pass `description` directly without the need to perform another dict lookup.